msal/src/main/java/com/microsoft/identity/client/controllers/MSALTokenOperation.java

```diff
@@ -0,0 +1,10 @@
+package com.microsoft.identity.client.controllers;
+
+import android.content.Intent;
+
+import com.microsoft.identity.client.AuthenticationResult;
+
+public interface MSALTokenOperation {
+    public AuthenticationResult execute();
+    public void notify(int requestCode, int resultCode, final Intent data);
```

Comment:
We may want to consider another name for this method: Java's [`Object#notify()`](https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#notify()) has special meaning around concurrency: I went looking in official docs for this method's behavior before I found its declaration in this PR